app/views/users/edit.html.haml

```diff
@@ -0,0 +1,29 @@
+= content_for :head_content do
+  :javascript
+    $(document).ready(function() {
+      $("input#user_email").keyup(function(e) {
+        $("input#user_username").val(this.value);
+      });
+
+      $("input#user_email").change(function(e) {
+        $("input#user_username").val(this.value);
+      });
+    });
+
+= content_for :h1 do
+  =h current_facility
+
+= content_for :sidebar do
+  = render "admin/shared/sidenav_users", sidenav_tab: "users"
+
+%h2= text("head", user: @user.full_name)
+%p= text("subhead")
+= simple_form_for(@user, url: facility_user_path(current_facility, @user), method: :put) do |f|
```

Comment:
Rails is actually smart enough to determine the `method` based on whether the `@user` is persisted or not. That also means that if you take out the `method` argument and copy it into `new_external` that form should work.